libvisual/tests/mem_test/mem_set_test.cpp

```diff
@@ -0,0 +1,68 @@
+#include "test.h"
+#include <cstdint>
+#include <libvisual/libvisual.h>
+#include <libvisual/lv_mem.h>
+#include <cstdint>
+#include <array>
+#include <algorithm>
+#include <vector>
+
+namespace {
+
+  void test_mem_set16 ()
+  {
+      constexpr std::uint16_t test_pattern = 0x1234;
+
+      constexpr auto test {[test_pattern] (auto x) {
+                               return x == test_pattern;
+                           }};
+
+      for (auto test_size = 1; test_size <= 128; test_size++) {
+          // Allocate extra elements in front and behind of test memory area.
+          std::vector<std::uint16_t> vec (test_size + 2);
+
+          visual_mem_set16 (&vec[1], test_pattern, test_size);
+
+          // Test for buffer overruns
+          LV_TEST_ASSERT (vec.front () == 0);
+          LV_TEST_ASSERT (vec.back () == 0);
+
+          // Test data area
+          LV_TEST_ASSERT (std::all_of (++begin (vec), --end (vec), test));
+      }
+  }
+
+  void test_mem_set32 ()
+  {
+      constexpr std::uint32_t test_pattern = 0x12345678;
+
+      constexpr auto test {[test_pattern] (auto x) {
+                               return x == test_pattern;
+                           }};
+
+      for (auto test_size = 1; test_size <= 128; test_size++) {
+          // Allocate extra elements in front and behind of test memory area.
+          std::vector<std::uint32_t> vec (test_size + 2);
+
+          visual_mem_set32 (&vec[1], test_pattern, test_size);
```

Comment:
@kaixiong I'd expect `test_size * sizeof(std::uint32_t)` instead of plain `test_size` here — I am missing something? One more case like that below.

Reply:
@kaixiong nevermind, I just figured what the `32` in the function name is for...